ext/reflection/php_reflection.c

```diff
@@ -3017,6 +3020,20 @@ ZEND_METHOD(reflection_method, __construct)
 		&& (mptr = zend_get_closure_invoke_method(Z_OBJ_P(orig_obj))) != NULL)
 	{
 		/* do nothing, mptr already set */
+	} else if (obj_handlers != NULL && obj_handlers->get_method != NULL) {
+		// a get_method object handler exists, so try to get the method from it
+		zend_string *method_string = zend_string_init(lcname, name_len, 0);
+
+		mptr = obj_handlers->get_method(&Z_OBJ_P(classname), method_string, NULL);
```

Comment:
Doesn't this throw for inaccessible methods?

Reply:
I haven't touched this in a while, zend_std_get_method would? What would be the best way to reconcile either letting get_method handle throwing an error or otherwise throwing the exception below?

I might be misunderstanding though 🙂